packages/front-end/components/Layout/TopNav.tsx

```diff
@@ -168,6 +168,13 @@ const TopNav: FC<{
                       e.preventDefault();
                       // @ts-expect-error TS(2722) If you come across this, please fix it!: Cannot invoke an object which is possibly 'undefin... Remove this comment to see the full error message
                       setOrgId(o.id);
+
+                      try {
+                        localStorage.setItem("gb-last-picked-org", `"${o.id}"`);
```

Comment:
we have `useLocalStorage` which handles all the try/catch and parsing

Reply:
i see. I think it might be nice to export a simple getter and setter function from useLocalStorage.ts and have the hook wrap this. That way all the parsing and try/catch is reusable.

Reply:
In my opinion on a matter that is related to React style/component architecture, an abstraction like `useLocalStorage()` should have an API like this:

```ts
const [getItem, setItem] = useLocalStorage()
```

And rather than a hook that creates new instances of the value and setItem function, we could have it be a hook that uses a context provider so there's only one safe localStorage set of functions throughout the app render and they all reference the single instance from the context.

Right now we currently have something like `useLocalStorageValue(key, defaultValue)` with a misleading name `useLocalStorage` because we do not actually get to use local storage, only read/write a single value in it. It also means if we need to get or set 10 items, we need to call this hook 10 times, which would be kind of annoying with the returned names, as we'd have to name the functions with what they're getting and setting, e.g. `const [lastPickedOrg, setLastPickedOrg] = useLocalStorageValue('gb-last-picked-org')`. It did not look like the right thing to use in this instance. 

I have an example of what I mean in the `useCopyToClipboard()` hook that returns a `performCopy` function that allows us to it with any value (e.g. `performCopy(theValue)`). I did not tie every call of the hook to a single value.